src/test/scala/CloudWatchMetricFilterSpec.scala

```diff
@@ -0,0 +1,39 @@
+
+import com.blacklocus.metrics.CloudWatchReporterBuilder
+import com.codahale.metrics.{Metric, MetricFilter, MetricRegistry}
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{FlatSpec, Matchers, PrivateMethodTester}
+import CloudWatchMetricFilter.buildMetricFilter
+import scala.util.matching.Regex
+import java.util.regex.PatternSyntaxException
+
+
+class CloudWatchMetricFilterSpec extends FlatSpec with Matchers with PrivateMethodTester with MockFactory {
+
+  private val metricRegistry: MetricRegistry = new MetricRegistry
+  private val metric: Metric = mock[Metric]
+
+
+  "buildMetricsFilter" should "match any metric when empty string filter is given" in {
+    val filter = buildMetricFilter(new Regex(""))
```

Comment:
Similar to above, you can use `"".r` instead of `new Regex("")` just looks a little bit nicer 😃